src/main/java/io/cdap/plugin/cloud/vision/FilePathSourceConfig.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cloud.vision;
+
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.Constants;
+import io.cdap.plugin.common.IdUtils;
+
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Defines a {@link PluginConfig} that File Path batch source can use.
+ */
+public class FilePathSourceConfig extends PluginConfig {
+
+  public static final Schema SCHEMA = Schema.recordOf("schema", Schema.Field.of(CloudVisionConstants.PATH_FIELD_NAME,
+                                                                                Schema.of(Schema.Type.STRING)));
+
+  @Name(Constants.Reference.REFERENCE_NAME)
+  @Description(Constants.Reference.REFERENCE_NAME_DESCRIPTION)
+  private String referenceName;
```

Comment:
There is `io.cdap.plugin.common.ReferencePluginConfig` which holds `referenceName`